dev-libs/libiconv/libiconv-1.15.recipe

```diff
@@ -0,0 +1,201 @@
+It provides support for these encodings:
+
+European languages
+- ASCII, ISO-8859-{1,2,3,4,5,7,9,10,13,14,15,16}, KOI8-R, KOI8-U, KOI8-RU, \
+CP{1250,1251,1252,1253,1254,1257}, CP{850,866,1131}, \
+Mac{Roman,CentralEurope,Iceland,Croatian,Romania}, \
+Mac{Cyrillic,Ukraine,Greek,Turkish}, Macintosh
+Semitic languages
+- ISO-8859-{6,8}, CP{1255,1256}, CP862, Mac{Hebrew,Arabic}
+Japanese
+- EUC-JP, SHIFT_JIS, CP932, ISO-2022-JP, ISO-2022-JP-2, ISO-2022-JP-1
+Chinese
+- EUC-CN, HZ, GBK, CP936, GB18030, EUC-TW, BIG5, CP950, BIG5-HKSCS, \
+BIG5-HKSCS:2004, BIG5-HKSCS:2001, BIG5-HKSCS:1999, ISO-2022-CN, ISO-2022-CN-EXT
+Korean
+- EUC-KR, CP949, ISO-2022-KR, JOHAB
+Armenian
+- ARMSCII-8
+Georgian
+- Georgian-Academy, Georgian-PS
+Tajik
+- KOI8-T
+Kazakh
+- PT154, RK1048
+Thai
+- ISO-8859-11, TIS-620, CP874, MacThai
+Laotian
+- MuleLao-1, CP1133
+Vietnamese
+- VISCII, TCVN, CP1258
+Platform specifics
+- HP-ROMAN8, NEXTSTEP
+Full Unicode
+- UTF-8
+- UCS-2, UCS-2BE, UCS-2LE
+- UCS-4, UCS-4BE, UCS-4LE
+- UTF-16, UTF-16BE, UTF-16LE
+- UTF-32, UTF-32BE, UTF-32LE
+- UTF-7
+- C99, JAVA
+Full Unicode, in terms of uint16_t or uint32_t (with machine dependent \
+endianness and alignment)
+- UCS-2-INTERNAL, UCS-4-INTERNAL
+Locale dependent, in terms of 'char' or 'wchar_t' (with machine dependent \
+endianness and alignment, and with OS and locale dependent semantics)
+- char, wchar_t
+- The empty encoding name \"\" is equivalent to \"char\": it denotes the \
+locale dependent character encoding.
+
+When configured with the option --enable-extra-encodings, it also provides \
+support for a few extra encodings:
+European languages
+- CP{437,737,775,852,853,855,857,858,860,861,863,865,869,1125}
+Semitic languages
+- CP864
+Japanese
+- EUC-JISX0213, Shift_JISX0213, ISO-2022-JP-3
+Chinese
+- BIG5-2003 (experimental)
+Turkmen
+- TDS565
+Platform specifics
+- ATARIST, RISCOS-LATIN1
+
+It can convert from any of these encodings to any other, through Unicode \
+conversion.
+It has also some limited support for transliteration, i.e. when a character \
+cannot be represented in the target character set, it can be approximated \
+through one or several similarly looking characters. Transliteration is \
+activated when \"//TRANSLIT\" is appended to the target encoding name.
+
+libiconv is for you if your application needs to support multiple character \
+encodings, but that support lacks from your system."
+HOMEPAGE="http://www.gnu.org/software/libiconv
+	http://www.gnu.org/software/iconv"
+COPYRIGHT="2000-2017 Free Software Foundation, Inc."
+LICENSE="GNU LGPL v2
+	GNU GPL v3	"
+REVISION="1"
+SOURCE_URI="http://ftp.gnu.org/pub/gnu/libiconv/libiconv-$portVersion.tar.gz"
+CHECKSUM_SHA256="ccf536620a45458d26ba83887a983b96827001e92a13847b45e4925cc8913178"
+
+ARCHITECTURES="x86_gcc2 x86 x86_64 arm"
+SECONDARY_ARCHITECTURES="x86"
+
+PROVIDES="
+	libiconv$secondaryArchSuffix = $portVersion compat >= 1.15
+	lib:libiconv$secondaryArchSuffix = 2.6.0 compat >= 2
```

Comment:
Shall we use `lib{iconv,charset}Version{,Compat}` and `portVersionCompat` ?
```diff
 ARCHITECTURES="x86_gcc2 x86 x86_64 arm"
 SECONDARY_ARCHITECTURES="x86"
+
+libiconvVersion="2.6.0"
+libcharsetVersion="1.0.0"
+
+portVersionCompat="$portVersion compat >= 1.15"
+libiconvVersionCompat="$libiconvVersion compat >= ${libiconvVersion%%.*}"
+libcharsetVersionCompat="$libcharsetVersion compat >= ${libcharsetVersion%%.*}"
+
 PROVIDES="
- 	libiconv$secondaryArchSuffix = $portVersion compat >= 1.15
+ 	libiconv$secondaryArchSuffix = $portVersionCompat
-	lib:libiconv$secondaryArchSuffix = 2.6.0 compat >= 2
+	lib:libiconv$secondaryArchSuffix = $libiconvVersionCompat
-	lib:libcharset$secondaryArchSuffix = 1.0.0 compat >= 1
+	lib:libcharset$secondaryArchSuffix = $libcharsetVersionCompat
	"
 if [ -z "$secondaryArchSuffix" ]; then
 	PROVIDES="$PROVIDES
-		cmd:iconv = 2.6.0 compat >= 2
+		cmd:iconv = $libiconvVersionCompat
 		"
 fi
```

Reply:
You could just drop `portVersionCompat`. The port version is already 1.15.

Reply:
Well, it might still be worth defining a `portVersionCompat` with `$portVersion` and the hardcoded `compat >= 1.15`, just in case 1.15.1 gets released.

Reply:
Most of the time we would be refering to either `lib:` or `devel:` versions, so does having a compat string for the port itself do anything good?

Reply:
I just made a quick search and found 3 recipes (all in `dev-vcs/subversion/`) which include `libiconv_devel >= 1.13` in their `BUILD_REQUIRES`:
* subversion-1.6.18.recipe
* subversion-1.8.11.recipe
* subversion-1.9.7.recipe

So I guess we should change these 3 recipes to replace that condition by an appropriate condition on `lib:libiconv` and `devel:libiconv`. (I guess ">= 2" would be OK.)

Reply:
Should be `>= 2.5.0`. That's ABI version correspond to version 1.13 of `iconv`

Reply:
@fbrosson From [BuildingPackages#VersionStrings][0], it seems that `compat` only serves the purpose of defining the oldest compatible version, not the one you could compare to. This effectively means you could just drop the `compat` string for the port and `libiconv_devel >= 1.13` would still be vaild.